### ISSUE TITLE
Manual refresh required for scripts to be processed

### ISSUE THREAD

**Issue by cast-away** [renyuneyun/Easer#159]
Hiya,
With my current setup on Android 8 (details below) I need to click on the refresh button on the outline page for the scripts to be triggered. Am I missing something?
profiles: 
- wifi on
- wifi off 

event:
- cell location (home)

scripts:
1. arrive home
- parent: empty (what would this do incidentally?)
- profile: wifi on
- predefined event: home
- repeatable when satisfied + persistent

2. leave home
- parent: empty
- profile: wifi off
- predefined event: home (reverse scenario)
- repeatable when satisfied + persistent

**Comment by cast-away** [comment]
Hiya,
Thanks for looking into this. Both scripts are correctly triggered when pressing the refreshing button but do nothing on their own.
I have changed them to use Conditions instead with the exact same result. Last time I gave it an hour after I arrived home, wifi was still off and turned on instantly after a manual refresh.

**Comment by renyuneyun** [comment]
That seems strange... Though your initial set of "persistent" should be changed to "volatile" (but this doesn't matter as you have changed to Condition).
After all, the change of cell tower is based solely on the information provided by Android, and should immediately be caught by Easer.

Is this the only problem you encountered in Easer? Does other Events or Conditions work?

I'll think about this more...
In the meantime, do you mind sharing your scripts&events&conditions&profiles? You can do this by using the "export" function provided in the settings page. Note this contains all your scripts&events&conditions&profiles so remember to remove sensitive information beforehands :P

**Comment by racuna** [comment]
Happened to me too:

1. Condition (working days)
1.1 Condition (before 6pm)
1.1.1 Event (after 9am) -> moderate sound volume
1.1.2 Condition (after 9am) -> moderate sound volume

Y tried both Event and Condition, both individually and now both together. Still, sometimes I have to open the app and press "Refresh".

**Comment by conceptrat** [comment]
I'm wondering if the "service" is getting snoozed/dozed somehow?

**Comment by andyvandermeyde** [comment]
It just does not work, I have the simplest setup ever. Just a condition with a time, it simply does not do anything until you press the refresh button. Not even if I'm in the app when the condition becomes true it's doing anything.
I don't now if it's an Android 8 issue but it's quite depressing.

**Comment by renyuneyun** [comment]
Thanks guys. I think I now know the problem...
The Service of loading Profile is not put to foreground (even after I put the other Service to foreground), and it may be killed/snoozed by the system...

Silly me, I'll fix this soon.

**Comment by andyvandermeyde** [comment]
Alright, thanks for investigating it and nice that you found the problem. If you let me know when there is an update I can test it with a couple of different devices.

**Comment by conceptrat** [comment]
@renyuneyun Was about to try installing your app to test out my theory above.  Yep I've never installed it but I'm making comments ;).

The reason I suggested the service getting snoozed is because the other app I've been using for a while now, Tasker, had the same issue way back near it's first release and had an option available to create a notification to keep the app from being put to sleep or killed by the system process.  This was usually only necessary for some Android versions.  It's possible that this issue has come back in another form in Android 8.

This is all just conjecture, as I'm not a fully competent Android developer yet.

**Comment by andyvandermeyde** [comment]
Hi @renyuneyun, 
thanks for the effort you put into updating the app. Unfortunately this still does not work at all. Even with the app open and in the foreground you are still required to press the update button manually, or to stop and re-start the whole thing from "Outline", for the conditions to be evaluated.

**Comment by renyuneyun** [comment]
Hi @andyvandermeyde ,
This seems strange, and I can't reproduce on two Android 8.1 devices. (Though, both devices have some strange Android behaviour...)
Can you share your setup? e.g. your Condition, Script (and Profile)?

p.s. if what you're encountering is purely for Time Event or Condition, it may have some delay because of Android's way of scheduling (for battery optimization).

**Comment by andyvandermeyde** [comment]
Hi! To make it even easier I recorded a video:
https://youtu.be/JnQz-bHp30U

At the end you can see that I click the update button just to show that the condition actually works, it's just never ever updating. I stopped after a few minutes but I could have left it there for hours and it would never update (I tried to leave it there for up to 24 hours).

**Comment by renyuneyun** [comment]
Hi @andyvandermeyde , Thanks for the video! This clarifies a lot, and it really seem to be a big problem.

Just one more info: does this happen only for Time (Event/Condition)? (Specifically, you could use battery related Event/Condition for a quick test.)

**Comment by andyvandermeyde** [comment]
Hi @renyuneyun ,
I think you might be onto something here. I tried the battery charging condition and this one seems to work properly. 
The problem could really be Time conditions...

**Comment by cast-away** [comment]
Hi,
Not sure this is limited to Time conditions as I am still struggling with location conditions: now reverse scenario works but not normal... Both have to be manually triggered still.
I have not yet found how to clean sensitive information from the log, apologies for this.

**Comment by chrm-software** [comment]
Hi,
I have the same problem with reverse condition for wifi connection. The normal conditions works (connecting to a defined wifi sets the correct profile), but leaving a wifi does not fire any event - not even hours later. Pressing refresh sets the correct profile.

**Comment by jmb20** [comment]
I am using Easer to run a script on boot (using status Connection "no connection" or "wireless", i.e. in any case) to 

1) activate Wifi (in case there is no connection), 
2) start Spotify, 
3) increase to maximum media volume over headphones. 

For some reason, the last step, i.e. increase to maximum media volume does not work consistently. It works if I manually refresh in the Easer main menu, but it does not work when the script is otherwise executed correctly (i.e. Wifi is activated, Spotify launched) at boot.

I have tried increasing volume to maximum both by using the corresponding Easer action as well as by running command "service call audio 3 i32 3 i32 15" on my rooted Android 7.1.2 device. It seems also like one cannot increase media volume to max in background, but only manually.

**Comment by andyvandermeyde** [comment]
> I am using Easer to run a script on boot (using status Connection "no connection" or "wireless", i.e. in any case) to
> 
> 1. activate Wifi (in case there is no connection),
> 2. start Spotify,
> 3. increase to maximum media volume over headphones.
> 
> For some reason, the last step, i.e. increase to maximum media volume does not work consistently. It works if I manually refresh in the Easer main menu, but it does not work when the script is otherwise executed correctly (i.e. Wifi is activated, Spotify launched) at boot.
> 
> I have tried increasing volume to maximum both by using the corresponding Easer action as well as by running command "service call audio 3 i32 3 i32 15" on my rooted Android 7.1.2 device. It seems also like one cannot increase media volume to max in background, but only manually in foreground.

I don't know anything about this but the first thing that comes to my mind is that some devices (like S8) have a warning message/popup that the user has to confirm manually when they try to set the maximum volume, it's pure speculation but maybe it has something to do with that.

**Comment by jmb20** [comment]
Indeed, this warning message is present when user tries to manually increase volume beyond a certain threshold and must be confirmed OK for volume to be increased beyond. I had hoped the Easer action or the above direct service call would circumvent the nag screen.

Since the volume change is performed when the script is manually triggered in Easer (by way of refresh) I wonder whether one can trigger an Easer script from a home screen icon button press? I don't necessarily need auto execution at boot, but I'd want Wifi connect, volume to max and Spotify launch to be a scripted sequence.

**Comment by urza9814** [comment]
Anyone ever find a resolution to this? I am having the same issue on my OnePlus 6 running LineageOS (no GApps) Android 11
